test/test/single_threaded_observable_vector.cpp

```diff
@@ -217,6 +217,22 @@ TEST_CASE("single_threaded_observable_vector")
         REQUIRE((vector_i.IndexOf(2, index) && index == 1));
         index = 0;
         REQUIRE((vector_o.IndexOf(box_value(2), index) && index == 1));
+
+        REQUIRE(false == vector_o.IndexOf(nullptr, index));
```

Comment:
```
REQUIRE(false == vector_o.IndexOf(box_value(L"hello"), index));
```

Looking for a string in a vector of integers should fail rather than crash.